### PR TITLE
Revert "Remove usages of Hadoop Path for Hive LocationService"

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Revert https://github.com/trinodb/trino/commit/8bd9f754fb2aec61ffa353adfb421b460b60d947.
Relates to https://github.com/trinodb/trino/pull/17804#discussion_r1231968909

## Release notes

(x) This is not user-visible or docs only and no release notes are required.

### CODE REVIEW

**Comment by martint** [comment]
Why revert? What would it take to fix the issue, instead?

cc @electrum

**Comment by findepi** [comment]
Hadoop `Path` doesn't support paths with double slashes.
Before https://github.com/trinodb/trino/commit/8bd9f754fb2aec61ffa353adfb421b460b60d947 Trino did not create Hive tables with such paths, sot he problem wasn't as exposed to users.

Since that commit we no longer normalize table location during CREATE TABLE, so Trino can easily create Hive tables with double slashes in the location, but these are not supported yet: https://github.com/trinodb/trino/pull/17804#discussion_r1229274970.

We should restore normalization to fix the problem and unblock the release.
Then we should probably reintroduce 8bd9f754fb2aec61ffa353adfb421b460b60d947 behind a config toggle, so that we can revisit all Hive connector places and fix (and test cover) all the changes related to de-hadoop-path-ing (https://github.com/trinodb/trino/issues/15921). Once all places are fixed, we should have disable table location normalization during CREATE TABLE by default.

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java [hunk 1]
```diff
@@ -1273,10 +1273,11 @@ private static String validateAvroSchemaLiteral(String avroSchemaLiteral)
         }
     }
 
-    private static Location getValidatedExternalLocation(String location)
+    private static Path getExternalLocationAsPath(String location)
     {
         try {
-            return Location.of(location);
+            Location.of(location); // Calling just for validation
```
Comment (ebyhr) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java:1279]: `testTableLocationTopOfTheBucket` doesn't pass without this line.

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/util/HiveWriteUtils.java [hunk 2]
```diff
@@ -452,8 +452,9 @@ public static Location getTableDefaultLocation(Database database, HdfsContext co
                 throw new TrinoException(HIVE_DATABASE_LOCATION_ERROR, format("Database '%s' location is not a directory: %s", schemaName, databasePath));
             }
         }
+        Location.of(databasePath.toString()); // Calling just for validation
```
Comment (ebyhr) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/util/HiveWriteUtils.java:455]: `testCreateSchemaWithIncorrectLocation` doesn't pass without this line.

**Comment by findepi** [comment]
https://github.com/trinodb/trino/pull/17958 may yield similar benfits (restore normalization), with fewer lines changed.

**Pull request closed without merging**